util/network-devp2p/src/discovery.rs

```diff
@@ -589,7 +598,7 @@ impl<'a> Discovery<'a> {
 			debug!(target: "discovery", "Found a known node in a bucket when processing discovery: {:?}/{:?}", known_node, node);
 			(known_node.address.endpoint == node.endpoint) && (known_node.last_seen.elapsed() < NODE_LAST_SEEN_TIMEOUT)
```

Comment:
I wonder if when we are in this case (and get `false`) we should still update the `node_buckets` when ping succeeds. Currently it's going to be cached in `other_observed_nodes`, but because it's also on the bucket list we will never get `true` for that node again.
Alternatively maybe change the order of checking the cache first and then node_buckets?